.github/workflows/vercel.yml

```diff
@@ -14,6 +14,8 @@ env:
 jobs:
   build_and_deploy:
     runs-on: ubuntu-latest
+    environment: ${{ inputs.env_name }} # Environment rules defined on GH UI
+    concurrency: ${{ inputs.env_name }} # Only one run per env at a time
```

Comment:
And this does what the comment says :)
Any other deployment on the same environment will be blocked until the lock is released